bin/oneoff/populate_discount_codes.rb

```diff
@@ -0,0 +1,112 @@
+# This script parses a CSV file of discount codes, and uses it to populate our
+# db. The CSV file(s) should not be checked into our repo as they contain secrets.
+# The file(s) are expected to have the following format:
+# 1 header row
+# 1 row for each entry, where a row contains a prefix column and a code column
+# The full code is prefix + code
+require 'securerandom'
+require 'csv'
+require_relative '../../dashboard/config/environment'
+require src_dir 'database'
+
+def populate_csv_data(filename, full_discount, expiration)
+  csv_contents = CSV.read(filename)
+  # skip header row
+  csv_contents.shift
+  csv_contents.each do |line|
+    prefix, code = line
+    CircuitPlaygroundDiscountCode.create!(
+      code: prefix + code,
+      full_discount: full_discount,
+      expiration: expiration
+    )
+  end
+end
+
+full_discount = ARGV[0]
+filename = ARGV[1]
+
+unless full_discount && filename
+  puts 'Expected usage: '
+  puts 'populate_discount_codes <full_discount?> <path_to_csv_file>'
+  exit(-1)
+end
+
+if full_discount == 'true'
+  full_discount = true
+elsif full_discount == 'false'
+  full_discount = false
+else
+  puts 'first argument must be true or false'
+  exit(-1)
+end
+
+populate_csv_data(filename, full_discount, Date.new(2018, 12, 31))
+
+puts 'Complete'
+
+#####
+# Begin debug helper methods
+#
+# What follows is a set of helper methods for discount code functionality that
+# can be copy/pasted into the console. They are designed to be used in non-prod
+# environments
+#####
+
+# This method is not used directly in this script, but can be used to populate
+# the db with fake discount codes. Usage looks something like:
+# populate_fake_data(10, false, Date.new(2018, 12, 31))
+# @param {boolean} full_discount - Is this a full or partial discount?
+# @param {number} n - How many entries to create
+def populate_fake_data(n, full_discount, expiration)
+  n.times do
+    # Create a bogus code
+    prefix = full_discount ? 'FAKE0_' : 'FAKE100_'
+    code = SecureRandom.hex(7)
+    CircuitPlaygroundDiscountCode.create!(
+      code: prefix + code,
+      full_discount: full_discount,
+      expiration: expiration
+    )
+  end
+end
+
+# This method is not used directly in this script, but can be used to add a teacher
+# to a CSD cohort, so that they will be considered PD eligible for discount codes.
+# Because this is modifying the cohort, it should not be run on prod (or if it is,
+# you need to be sure to clean up after yourself).
+def set_user_pd_eligible(teacher)
+  teacher.cohorts << Cohort.find_or_create_by('CSD-TeacherConHouston')
```

Comment:
Should this method raise if the environment is prod?

Reply:
I'm not sure that it's necessary. The nature of these scripts is already such that people won't just randomly be running them. I think it's safe to assume that whoever is running them has a fair understanding of what they're doing.

I could see there being some possibility that we do want to run in prod, but be very careful about cleaning up after ourselves.

Reply:
Though I could be convinced otherwise pretty easily if you feel strongly.

Reply:
Just figured if it's easy to do, better safe than sorry.  It looks like you're not including all of our helpers around that stuff though.  Don't worry about it.